liquify handles

TurboGit opened the pull request:
better control over the liquify nodes at any resolution/zoom.

TurboGit commented:
Please, test this last version makes the liquify controls far more fluid to use.

aurelienpierre commented on src/iop/liquify.c:
```diff
@@ -2268,9 +2183,219 @@ static float find_nearest_on_curve_t(const float complex p0,
+          if(t > 0.0f && t < 1.0f)
+          {
+            const float complex linepoint = cmix(lp1, lp2, t);
+            const float d = cabsf(linepoint - *pt);
```
d should be a euclidean distance, no ? aka hypotf(x_cursor - x_node, y_cursor - y_node).

TurboGit replied:
if I'm not mistaken this is what cabsf() does here.

da-phil replied:
The `cabs(z)` function is equivalent to `hypot(z.x, z.y)`, so you're both right :smile:

aurelienpierre replied:
I'm just concerned that dealing with complex numbers will be slower than dealing with a vector of floats

TurboGit replied:
No, behind the scene this is just a set of two float. Nothing more, nothing less and all nice routines coming with them are very useful in a two space canvas.

rawfiner commented:
Tested quickly, much better indeed! Thanks!

TurboGit commented:
Thanks @rawfiner for testing. Indeed much better too. I suppose this will help even more on slow computer has now moving the forms is really light on the CPU.

aurelienpierre commented:
Code looks good, feature works much better, thanks !

2 minor comments, though :
1. to drag the vector, one has to drag the point of the arrow, while we would expect to interact with the center of gravity of the arrow
2. on HighDPI (×2) screens, you need to be super accurate on node selection. Maybe making the radius of sensivitity proportional to DPI factor (`DT_PIXEL_APPLY_DPI()` macro) should help.

TurboGit commented:
>     1. to drag the vector, one has to drag the pointy head of the arrow, while we would expect to interact with the center of gravity of the arrow,

fixed.

>     2\. on HighDPI (×2) screens, you need to be super accurate on node selection. Maybe making the radius of sensivitity proportional to DPI factor (`DT_PIXEL_APPLY_DPI()` macro) should help.

Fixed too.

Can you double check and report ?

aurelienpierre commented:
It works beautifully, thanks a lot !

TurboGit commented:
Merged in master and queued for 3.4.1.
